fv3config/_datastore.py

```diff
@@ -125,51 +133,50 @@ def gsutil_is_installed():
 
 
 def check_if_data_is_downloaded():
-    if not os.path.isdir(local_archive_dir):
+    if not os.path.isdir(get_cache_dir()) or len(os.listdir(get_cache_dir())) == 0:
         raise DataMissingError(
             f'Required data for running fv3gfs not available. Try python -m fv3config.download_data or ensure_data_is_downloaded()'
         )
 
 
 def ensure_data_is_downloaded():
     """Check of the cached data is present, and if not, download it."""
-    if not os.path.isfile(local_archive_filename):
-        download_data_archive()
-    if not os.path.isdir(local_archive_dir):
-        extract_data()
+    os.makedirs(get_cache_dir(), exist_ok=True)
+    if len(os.listdir(get_cache_dir())) == 0:
+        with tempfile.NamedTemporaryFile(mode='wb') as archive_file:
+            download_data_archive(archive_file)
+            archive_file.flush()
+            extract_data(archive_file.name)
 
 
 def refresh_downloaded_data():
     """Delete the cached data (if present) and re-download it."""
-    os.remove(local_archive_filename)
-    shutil.rmtree(app_data_dir)
+    shutil.rmtree(get_cache_dir())
```

Comment:
I always wonder how to protect something like that from cleaning away my harddrive in case get_cache_dir() returns something whacky. Can we impose some safety here? This probably holds true for all rmtree() calls in the code.

Reply:
I think we've tried to put as many warnings as we can on the documentation of `refresh_downloaded_data` about it deleting cached files. There are unit tests on this functionality, so the main risk is that someone sets the cache dir to something wacky, but now we're also appending `fv3config-cache` to the directory they set (or to the environment variable), so I think we're pretty safe.

On top of *all* that (remembering that this is a private function, which can only be accessed through `python3 -m fv3config.refresh-data`), I've added some interactive statements when users run the refresh command that will warn them this directory may be deleted, and given an opportunity to cancel. We don't have any current use case that requires the refresh to be non-interactive.

To some extent, the only way to be completely safe against wiping your data is backups. Barring that, I think we're now at a pretty high level of safety.